### PR TITLE
test: add beta scale test

### PR DESCRIPTION
**Pull request by njtran**
Fixes #N/A 

**Description**
This adds in v1beta1 Scale testing and adds in a alpha-e2e-scale trigger that would trigger the Alpha scale tests. 
**How was this change tested?**
- make verify
- make e2etests

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/alpha-e2e-scale-trigger.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: E2EScaleTrigger
```
Comment (jonathan-innis) [inline .github/workflows/alpha-e2e-scale-trigger.yaml:1]: Are you allowed to have two workflows with the same name? I think this breaks?

File: .github/workflows/alpha-e2e-scale-trigger.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  workflow_run:
+    workflows: [ApprovalComment]
+    types: [completed]
+  workflow_dispatch:
+jobs:
+  resolve:
+    if: (github.repository == 'aws/karpenter' && (github.event_name != 'workflow_run' || github.event.workflow_run.conclusion == 'success')) || github.event_name == 'workflow_dispatch'
+    uses: ./.github/workflows/resolve-args.yaml
+    with:
+      allowed_comment: "scale"
```
Comment (jonathan-innis) [inline .github/workflows/alpha-e2e-scale-trigger.yaml:14]: Does this mean that a `/karpenter scale` comment will trigger both workflows? Is that correct?

File: test/suites/beta/scale/deprovisioning_test.go [hunk 3]
```diff
@@ -0,0 +1,788 @@
...
+var _ = Describe("Deprovisioning", Label(debug.NoWatch), Label(debug.NoEvents), func() {
+	var nodePool *corev1beta1.NodePool
+	var nodeClass *v1beta1.EC2NodeClass
+	var deployment *appsv1.Deployment
+	var deploymentOptions test.DeploymentOptions
+	var selector labels.Selector
+	var dsCount int
+
+	BeforeEach(func() {
+		env.ExpectSettingsOverriddenLegacy(map[string]string{"featureGates.driftEnabled": "true"})
```
Comment (jonathan-innis) [inline test/suites/beta/scale/deprovisioning_test.go:82]:
```suggestion
		env.ExpectSettingsOverridden(v1.EnvVar{Name: "FEATURE_GATES", Value: "Drift=True"})
```

Probably want to use the updated way of upserting the options

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by engedaam — APPROVED** [review]
LGTM 🚀

**Pull request merged**